cartridge-cli.lua

```diff
@@ -682,19 +682,25 @@ end
 
 local function normalize_version(str)
     local patterns = {
+        "(%d+)%.(%d+)%.(%d+)-(%d+)-(%g+)",
         "(%d+)%.(%d+)%.(%d+)",
         "(%d+)%.(%d+)",
         "(%d+)"
     }
 
     for _, pattern in ipairs(patterns) do
-        local major, minor, patch = string.match(str, pattern)
-
+        local major, minor, patch, commit_num, commit_hash = string.match(str, pattern)
+        local commit
         if major ~= nil then
             minor = minor or '0'
             patch = patch or '0'
-
-            return {major, minor, patch}
+            if commit_num ~= nil then
+                commit = tostring(commit_num)
+            end
+            if commit_hash ~= nil then
+                commit = commit .. '-' .. tostring(commit_hash)
+            end
+            return {major, minor, patch}, commit
```

Comment:
Maybe it's time to get rid of this horrible multi-return.
We already discussed that normalized version can be a string and I hope it will.

BTW, in the code `release` name is used to save `commit` value, it confuses a little.

Reply:
Replace `{major, minor, patch` to one "major.minor.patch" string

Change "commit" to "release"

Reply:
Also may be we can return simply "version-release"... But this code is quite fragile, I don't want to do it in this patch